### PR TITLE
Update auth to try auth on foreground and conversation list open

### PR DESCRIPTION
**Pull request by brianmcdon**
@eoinoconnell this PR will stop issues arising from the first setup/register attempt failing. Now both setup and register will be tried on foreground or conversation list open as necessary.

### CODE REVIEW

File: Drift/Managers/ReachabilityManager.swift [hunk 1]
```diff
@@ -17,9 +17,9 @@ public extension Notification.Name {
 
 class ReachabilityManager {
     static var sharedInstance: ReachabilityManager = ReachabilityManager()
-    let networkReachabilityManager = Alamofire.NetworkReachabilityManager()
+    let networkReachabilityManager = Alamofire.NetworkReachabilityManager(host: "www.apple.com")
```
Comment (eoinoconnell) [inline Drift/Managers/ReachabilityManager.swift:20]: Whats the idea behind this @brianmcdon ?

File: Drift/Public/Drift.swift [hunk 2]
```diff
@@ -19,6 +19,7 @@ open class Drift: NSObject {
      
     */
     open class func setup(_ embedId: String) {
+        DriftDataStore.sharedInstance.setEmbedId(embedId)
```
Comment (eoinoconnell) [inline Drift/Public/Drift.swift]: Do we want to do this? In the retrieve data from embeds call we check to make sure the embed is the same as whats in the data store to drop all the caches and start again.

File: Drift/Public/Drift.swift [hunk 3]
```diff
@@ -30,6 +31,8 @@ open class Drift: NSObject {
      
     */
     open class func registerUser(_ userId: String, email: String) {
+        DriftDataStore.sharedInstance.setUserId(userId)
+        DriftDataStore.sharedInstance.setEmail(email)
```
Comment (eoinoconnell) [inline Drift/Public/Drift.swift]: Maybe move this to inside the register User method

File: Drift/Public/Drift.swift (hunk 2 above)
Comment (brianmcdon) [reply Drift/Public/Drift.swift]: 👍 we don't need this we already have what we need through the embed itself.

**Review by eoinoconnell — APPROVED** [review]

**Pull request merged**